dm/master/scheduler/scheduler.go

```diff
@@ -384,6 +405,111 @@ func (s *Scheduler) GetSourceCfgByID(source string) *config.SourceConfig {
 	return &clone
 }
 
+// transferWorkerAndSource transfer worker and source bound relations.
+// lworker, "", "", rsource				This means a unbounded source bounded to a free worker
+// lworker, lsource, rworker, "" 		This means transfer a source from a worker to another free worker
+// lworker, lsource, "", rsource		This means transfer a worker from a bounded source to another unbounded source
+// lworker, lsource, rworker, rsource	This means transfer two bounded relations.
+func (s *Scheduler) transferWorkerAndSource(lworker, lsource, rworker, rsource string) error {
+	var (
+		lw           *Worker
+		rw           *Worker
+		boundWorkers []string
+		bounds       []ha.SourceBound
+		bound1       ha.SourceBound
+		bound2       ha.SourceBound
+	)
+
+	s.logger.Info("transfer source and worker", zap.String("left worker", lworker), zap.String("left source", lsource), zap.String("right worker", rworker), zap.String("right source", rsource))
+
+	if lworker != "" {
+		lw = s.workers[lworker]
+	}
+	if rworker != "" {
+		rw = s.workers[rworker]
+	}
+
+	// get current bounded workers.
+	if lworker != "" && lsource != "" {
+		boundWorkers = append(boundWorkers, lworker)
```

Comment:
seems we could simplify the arguments, `lsource` equals `lw.Bound().Source`

Reply:
There is a case that lworker is nil and lsource not nil. e.g. lsource is unbounded, we want to bound it with another worker.